### PR TITLE
WIP - Build basic GUI for a Unicore JupyterLab extension

### PR DESCRIPTION
**Pull request by liadomide**
We intend to have a JupyterLab extension, with a table as front-end

### CODE REVIEW

**Comment by davidbacter01** [comment]
Regarding the yarn.lock and package-lock.json, I have found that it is recommended to commit this files. This thread on stackoverflow talks in more detail about why it is recommended: https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5

File: src/index.ts [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
   JupyterFrontEndPlugin
 } from '@jupyterlab/application';
 
+import {
+  ICommandPalette,
+  MainAreaWidget,
+  WidgetTracker
+} from '@jupyterlab/apputils';
+
+import { PyunicoreWidget, DataType } from "./pyunicoreWidget";
```
Comment (liadomide) [inline src/index.ts:13]: Here we should pay attention to the server checks (e.g. user single quote) and implement those suggestions.

Comment (liadomide) [inline src/index.ts:13]: Here we should pay attention to the server checks (e.g. user single quote) and implement those suggestions.

**Comment by liadomide** [comment]
I close this as #2 takes over and includes these changes

**Pull request closed without merging**